src/compare_tracker_with_nxn_sheet.py

```diff
@@ -17,53 +17,207 @@
+    response = rq.get(search_url)
+    response_json = response.json()
+    result_list = response_json['resultList']['result']
+    result_list = [result for result in result_list if result['pubType'] != 'preprint']
+    response_json['resultList']['result'] = result_list
+    response_json['hitCount'] = len(result_list)
+    return response_json['resultList']['result'][0]['pmid'] if response_json['hitCount'] == 1 else ""
+
+
+
+def set_tissue(value):
+    if not value:
+        return ""
+    tissue_source_list = []
+    for tissue in value.split(','):
+        if "culture" in tissue.lower():
+            tissue_source_list.append("Cell line")
+        elif "organoid" in tissue.lower():
+            tissue_source_list.append('Organoid')
+        else:
+            tissue_source_list.append('Primary')
+    tissue_source_list = list(set(tissue_source_list))
+    return ','.join(tissue_source_list)
+
+
+def set_organ(value):
+    if not value:
+        return ""
+    value = value.split('\t')
+    return value[0] if value[0] else value[1]
+
+
+def find_header_index(matrix: [[]], value: str, header_row_index: int = 0):
+    header_row = matrix[header_row_index]
+    index_value = header_row.index(value)
+    if index_value == -1:
+        raise ChangedHeaders(value)
+    return index_value
+
+def replace_all_values(value, valentines_database, row):
+    while value.find('{') != -1:
+        start_index = value.find('{')
+        end_index = value.find('}')
+        valentines_header_value = value[start_index + 1:end_index]
+        v_index = find_header_index(valentines_database, valentines_header_value)
+        value = value.replace("{" + valentines_header_value + "}", row[v_index])
+    return value
+
+def eval_value(value, valentines_database, row):
+    if value.startswith('=='):
+        value = value[2:]
+        return replace_all_values(value, valentines_database, row)
+    if value.startswith('='):
+        value = value[1:]
+        value = replace_all_values(value, valentines_database, row)
+        return eval(value)
+
+
+def select_unique_studies(valentines_sheet: [[]], tracking_sheet: [[]]):
+    """
+    Filter unique studies based on:
+        - DOI
+        - Accession
+        - Paper title
+    :param valentines_sheet: [[]]
+                             Matrix representing nxn's database
+    :param tracking_sheet: [[]]
+                           Matrix representing the Dataset Tracking Sheet
+    :return:
+    """
+    # Retrieve the indexes from the headers
+    v_doi_index = find_header_index(valentines_sheet, 'DOI')
+    t_doi_index = find_header_index(tracking_sheet, 'doi')
+    v_data_location_index = find_header_index(valentines_sheet, 'Data location')
+    t_data_location_index = find_header_index(tracking_sheet, 'data_accession')
+    v_title_index = find_header_index(valentines_sheet, 'Title')
+    t_title_index = find_header_index(tracking_sheet, 'pub_title')
+
+
```

Comment:
@ESapenaVentura and I discussed this, but when deciding unique entries it might be good to check each of doi, accession and title with and ANY logical condition. i.e. if ANY of doi | accession | title are already found in the tracker, DONT add the row to the 'new row output'. Though we also discussed maybe we should still know about these to perhaps rectify or update either our tracker or Valentine's database